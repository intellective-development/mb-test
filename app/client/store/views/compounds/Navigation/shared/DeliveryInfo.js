// @flow
import * as React from 'react';
import _ from 'lodash';
import { connect } from 'react-redux';
import * as Ent from '@minibar/store-business/src/utils/ent';
import I18n from 'store/localization';
import { address_selectors } from 'store/business/address';
import { supplier_helpers, supplier_selectors } from 'store/business/supplier';
import type { Address } from 'store/business/address';
import type { DeliveryMethod, DeliveryMethodType } from 'store/business/delivery_method';

import { MBLoader, MBText } from '../../../elements';
import { MultiDeliveryMethodIcon } from '../../DeliveryMethodIcon';
import styles from './DeliveryInfo.scss';

type DeliveryInfoProps = {|
  current_address: Address,
  selected_delivery_methods: DeliveryMethod[],
  address_fetching: boolean
|};

const DeliveryInfo = ({current_address, selected_delivery_methods, address_fetching}: DeliveryInfoProps) => {
  let delivery_method_types = selected_delivery_methods.map(dm => dm.type);
  if (_.isEmpty(delivery_method_types)){
    delivery_method_types = ['on_demand'];
  }

  let content;
  if (current_address){
    content = <MBText.Span className={styles.cmNavDeliveryInfo_Address}>{formatAddress(current_address)}</MBText.Span>;
  } else if (address_fetching){
    content = <MBLoader />;
  } else {
    content = <MBText.Span className={styles.cmNavDeliveryInfo_AddAddress}>Add delivery address</MBText.Span>;
  }

  return (
    <div className={styles.cmNavDeliveryInfo_Container}>
      <MultiDeliveryMethodIcon
        width={24}
        height={24}
        delivery_method_types={delivery_method_types} />
      <div>
        <MBText.Span className={styles.cmNavDeliveryInfo_TextTitle}>
          {deliveryMessage(delivery_method_types)}
        </MBText.Span><br />
        {content}
      </div>
    </div>
  );
};


const DeliveryInfoSTP = () => {
  const findAddress = Ent.find('address');
  const findSuppliers = Ent.find('supplier');
  const findDeliveryMethods = Ent.find('delivery_method');

  return (state) => {
    const current_suppliers = findSuppliers(state, supplier_selectors.currentSupplierIds(state));
    const display_suppliers = supplier_helpers.displaySuppliers(current_suppliers);

    const selected_delivery_method_ids = display_suppliers.map(s => supplier_selectors.supplierSelectedDeliveryMethod(state, s.id));
    const selected_delivery_methods = findDeliveryMethods(state, selected_delivery_method_ids);

    return {
      current_address: findAddress(state, address_selectors.currentDeliveryAddressId(state)),
      address_fetching: address_selectors.isFetching(state),
      selected_delivery_methods
    };
  };
};
const DeliveryInfoContainer = connect(DeliveryInfoSTP)(DeliveryInfo);

export default DeliveryInfoContainer;

// utils

const formatAddress = (address: Address) => {
  if (!address.address2) return address.address1;
  return `${address.address1}, ${address.address2}`;
};

const deliveryMessage = (delivery_method_types: DeliveryMethodType[]) => {
  const uniq_types = _.uniq(delivery_method_types);
  const icon_name = _.sortBy(uniq_types).join('__');

  return I18n.t(`ui.nav.delivery_info.${icon_name}`);
};

export const __private__ = {
  deliveryMessage
};
