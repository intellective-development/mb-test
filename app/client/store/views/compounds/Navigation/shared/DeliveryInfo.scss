@import "~client/store/views/style/variables";

.cmNavDeliveryInfo_Container {
  display: flex;
  align-items: center;
}

.cmNavDeliveryInfo_TextTitle {
  @include font-size(12);
  color: $brand-medium-grey;
  letter-spacing: 0.4px;
}

.cmNavDeliveryInfo_AddAddress,
.cmNavDeliveryInfo_Address {
  @include font-size(13);
  @include overflow-ellipsis;

  display: inline-block;
  font-weight: $heavy-font-weight;
  margin-top: 4px;

  max-width: 160px;
}

.cmNavDeliveryInfo_AddAddress {
  color: $brand-red;
}
