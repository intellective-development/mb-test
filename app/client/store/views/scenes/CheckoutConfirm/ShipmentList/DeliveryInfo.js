// @flow

import * as React from 'react';
import { delivery_method_helpers } from 'store/business/delivery_method';

import { MBText } from '../../../elements';

type DeliveryInfoProps = {shipment: Object};
class DeliveryInfo extends React.PureComponent<DeliveryInfoProps> {
  render(){
    const { delivery_method } = this.props.shipment;

    return (
      <div className="csl__delivery__container">
        <div className="csl__delivery__name-container">
          <MBText.Span className="csl__delivery__name">
            {delivery_method_helpers.displayName(delivery_method)}
            <ClosedWarning delivery_method={this.props.shipment.delivery_method} />
          </MBText.Span>
        </div>
        <MBText.Span className="csl__delivery__next-delivery">
          {delivery_method_helpers.formatNextDelivery(delivery_method)}
        </MBText.Span>
      </div>
    );
  }
}

const ClosedWarning = ({delivery_method}) => {
  if (!delivery_method_helpers.isClosed(delivery_method)) return null;

  return <MBText.Span className="csl__delivery__closed">&ensp;Closed</MBText.Span>;
};

export default DeliveryInfo;
