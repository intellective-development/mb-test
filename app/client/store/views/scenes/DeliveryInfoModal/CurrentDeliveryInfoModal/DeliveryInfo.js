// @flow

import * as React from 'react';
import _ from 'lodash';
import { connect } from 'react-redux';
import * as Ent from '@minibar/store-business/src/utils/ent';
import { delivery_method_helpers } from 'store/business/delivery_method';
import { supplier_selectors, supplier_helpers } from 'store/business/supplier';
import type { Supplier } from 'store/business/supplier';

import AddressSection from './AddressSection';
import SupplierRow from './SupplierRow';
import { MBModal, MBButton } from '../../../elements';

type DeliveryInfoProps = {
  current_suppliers: Array<Supplier>,
  supplier_fetch_loading: boolean,
  deliveryInfoRouteTo: Function,
  hideModal: Function
};
const DeliveryInfo = ({current_suppliers, hideModal, supplier_fetch_loading, deliveryInfoRouteTo}: DeliveryInfoProps) => {
  return (
    <div>
      <MBModal.SectionHeader
        renderRight={() => <MBModal.Close onClick={hideModal} />} >
        Delivery Address
      </MBModal.SectionHeader>
      <AddressSection deliveryInfoRouteTo={deliveryInfoRouteTo} />
      <div className="currdel__done-container">
        <MBButton expand onClick={hideModal}>Done</MBButton>
      </div>
      <SupplierList
        suppliers={current_suppliers}
        supplier_fetch_loading={supplier_fetch_loading}
        deliveryInfoRouteTo={deliveryInfoRouteTo} />
    </div>
  );
};
const DeliveryInfoSTP = () => {
  const findSuppliers = Ent.query(Ent.find('supplier'), Ent.join('delivery_methods'));
  return (state) => {
    const current_supplier_ids = supplier_selectors.currentSupplierIds(state, { ignore_lazy_loaded: true });
    const current_suppliers = supplier_helpers.primarySuppliers(findSuppliers(state, current_supplier_ids));

    return {
      current_suppliers,
      supplier_fetch_loading: supplier_selectors.fetchLoading(state)
    };
  };
};
const DeliveryInfoContainer = connect(DeliveryInfoSTP)(DeliveryInfo);

// TODO: better splitting by type? right now we're just assuming there's only going to be 1 per type
const SupplierList = ({suppliers, supplier_fetch_loading, deliveryInfoRouteTo}) => {
  if (_.isEmpty(suppliers)) return null;
  let supplier_content;

  if (supplier_helpers.hasSplitSupplierTypes(suppliers)){
    const grouped_suppliers = _.groupBy(suppliers, 'type') || {};
    // if we have split types, we render them out as a list
    supplier_content = (
      _.values(grouped_suppliers).map(supplier_group => {
        const sortedGroup = _.sortBy(supplier_group.map(supplier => {
          return supplier ? {...supplier, open: delivery_method_helpers.isClosed(supplier.delivery_methods[0])} : {};
        }), ['open', 'name']);
        return (
          <div>
            { sortedGroup ? <MBModal.SectionHeader top={false}>{sortedGroup[0].type} Stores</MBModal.SectionHeader> : null }
            <div className="currdel__sr-row">
              { sortedGroup.map(supplier => (
                <SupplierRow
                  supplier={supplier}
                  deliveryInfoRouteTo={deliveryInfoRouteTo} />
              ))}
            </div>
          </div>
        );
      })
    );
  } else {
    // otherwise, we just render it out as a single supplier
    const supplier = suppliers[0];
    supplier_content = (
      <div>
        <MBModal.SectionHeader top={false}>Your Local Store</MBModal.SectionHeader>
        <div className="currdel__sr-row">
          <SupplierRow
            supplier={supplier}
            deliveryInfoRouteTo={deliveryInfoRouteTo} />
        </div>
      </div>
    );
  }

  return (
    <div className="currdel__sl__container">
      {supplier_content}
      <SupplierListLoader is_hidden={!supplier_fetch_loading} />
    </div>
  );
};

const SupplierListLoader = ({is_hidden}) => {
  if (is_hidden) return null;

  return <div className="currdel__sl__loading-overlay" />;
};

export default DeliveryInfoContainer;
