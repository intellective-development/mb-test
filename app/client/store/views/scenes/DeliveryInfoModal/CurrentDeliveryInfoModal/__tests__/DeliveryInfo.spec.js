import React from 'react';

import address_factory from 'client/store/business/address/__tests__/address.factory';
import supplier_factory from 'client/store/business/supplier/__tests__/supplier.factory';
import TestProvider from 'store/views/__tests__/utils/TestProvider';
import DeliveryInfo from '../DeliveryInfo';

describe('DeliveryInfo', () => {
  it('renders', () => {
    const address_norm = address_factory.normalize(address_factory.build());
    const supplier_norm = supplier_factory.normalize([supplier_factory.build()]);

    const initial_state = {
      address: {
        by_id: {
          ...supplier_norm.entities.address,
          ...address_norm.entities.address
        },
        current_delivery_address_id: address_norm.result
      },
      supplier: {
        by_id: supplier_norm.entities.supplier,
        current_ids: supplier_norm.result
      }
    };

    expect(render(
      <TestProvider initial_state={initial_state}>
        <DeliveryInfo />
      </TestProvider>
    )).toMatchSnapshot();
  });
});
