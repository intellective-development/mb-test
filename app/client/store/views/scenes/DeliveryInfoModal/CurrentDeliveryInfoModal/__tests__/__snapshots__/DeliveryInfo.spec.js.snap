// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeliveryInfo renders 1`] = `
<div>
  <div
    class="elMBModal_SectionHeader elMBModal_SectionHeader_Top"
  >
    <div
      class="elMBModal_SectionHeader_Filler"
    />
    <h4
      class="elMBText elMBText_ResetSpacing elMBModal_SectionHeaderText"
    >
      Delivery Address
    </h4>
    <div
      aria-label="Close"
      class="elMBModal_Close"
    >
      <img
        alt="close"
        class="el-mbicon--close"
        src="/assets/components/elements/mb-icon/close.png"
        srcset="/assets/components/elements/mb-icon/close@2x.png 2x, /assets/components/elements/mb-icon/close@3x.png 3x"
      />
    </div>
  </div>
  <div
    class="currdel__address__container"
  >
    <div
      class="cm-ae-gift__container"
    >
      <img
        alt="Gift"
        class="cm-ae-gift__icon"
        src="/assets/sprite_sources/icons/gift_icon.png"
      />
      <span
        class="cm-ae-gift__text"
      >
        Sending a gift? Use the recipient's address.
      </span>
    </div>
    <div
      class="cm-ae-container"
      id="address-entry"
    >
      <div
        class="cm-ae-error"
      >
        <p
          class="cm-ae-error__text cm-ae-error__text--empty"
        >
          An unexpected error has occured.
        </p>
      </div>
      <div
        class="cm-ae-input__row"
      >
        <div
          class="cm-ae-input__container"
        >
          <img
            alt="pin"
            class="el-mbicon--pin cm-ae-input__pin_icon"
            src="/assets/components/elements/mb-icon/pin.png"
            srcset="/assets/components/elements/mb-icon/pin@2x.png 2x, /assets/components/elements/mb-icon/pin@3x.png 3x"
          />
          <input
            autocomplete="off"
            class="elMBInput cm-ae-input"
            id="addres-input"
            name="addres-input"
            placeholder="Enter a delivery addâ€‹ress to shop"
            required=""
            type="text"
            value=" 123 Jump Street, Apt. 10, New York New York 12345"
          />
          <img
            alt="clear"
            class="el-mbicon--clear cm-ae-input__close_button"
            src="/assets/components/elements/mb-icon/clear.png"
            srcset="/assets/components/elements/mb-icon/clear@2x.png 2x, /assets/components/elements/mb-icon/clear@3x.png 3x"
          />
          <span
            class="cm-ae-input__loader button_hidden"
          />
          <div
            class="cm-ae-dropdown__container"
          >
            <ul
              class="cm-ae-dropdown__list"
            />
          </div>
        </div>
        <button
          class="elMBButton elMBButton__Default elMBButton__Tall cm-ae-button button_hidden"
          disabled=""
          id="address-button"
        >
          Go
          <div
            class="button__loader"
          />
        </button>
      </div>
      <div
        id="google_places_target"
      />
    </div>
  </div>
  <div
    class="currdel__done-container"
  >
    <button
      class="elMBButton elMBButton__Default elMBButton__Medium elMBButton__Expand"
    >
      Done
    </button>
  </div>
  <div
    class="currdel__sl__container"
  >
    <div>
      <div
        class="elMBModal_SectionHeader"
      >
        <div
          class="elMBModal_SectionHeader_Filler"
        />
        <h4
          class="elMBText elMBText_ResetSpacing elMBModal_SectionHeaderText"
        >
          Your Local Store
        </h4>
        <div
          class="elMBModal_SectionHeader_Filler"
        />
      </div>
      <div
        class="currdel__sr-row"
      >
        <div
          class="currdel__sr__container"
        >
          <div
            class="currdel__sr__header"
          >
            <div
              class="currdel__sr__supplier"
            >
              <h4
                class="elMBText elMBText_ResetSpacing currdel__supplier-name"
              >
                Manhattan Wine
              </h4>
              <div
                class="el-touchable el-touchable--disabled"
                role="button"
                tabindex="-1"
              >
                <h5
                  class="elMBText elMBText_ResetSpacing currdel__supplier-location"
                >
                   123 Jump Street, New York, New York
                </h5>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
